src/test/java/ui/main_menu/MainMenuTest.java

```diff
@@ -0,0 +1,120 @@
+package ui.main_menu;
+
+import game.ui.SceneLauncher;
+import game.ui.game.GameScene;
+import game.ui.menu.MenuScene;
+import org.assertj.swing.fixture.FrameFixture;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import ui.util.TestComponentNames;
+
+import static org.junit.Assert.assertEquals;
+
+public class MainMenuTest {
+    private FrameFixture window;
+    private MenuScene menuScene;
+    private SceneLauncher launcher;
+
+    @Before
+    public void setUp() {
+        launcher = new SceneLauncher();
+        menuScene = (MenuScene)launcher.getCurrentScene();
+        window = new FrameFixture(menuScene);
+        window.show();
+    }
+
+    @Test
+    public void startGameWithZeroPlayers(){
+        //ARRANGE
+
+        //ACT
+        window.button(TestComponentNames.START_GAME).click();
+
+        //ASSERT
+        assertEquals( launcher.getCurrentScene(), menuScene);
+    }
+
+    @Test
+    public void startGameWithAddingPlayerWithTooLongName(){
+        //ARRANGE
+        var TOO_LONG_NAME = "12345678901";
+
+        //ACT
+        window.button(TestComponentNames.NEW_PLAYER).click();
+        var optionPaneFixture = window.optionPane();
+        optionPaneFixture.textBox().enterText(TOO_LONG_NAME);
+        optionPaneFixture.okButton().click();
+        var optionPaneTooLong = window.optionPane();
+        optionPaneTooLong.okButton().click();
+        window.button(TestComponentNames.START_GAME).click();
+
+        //ASSERT
+        assertEquals(launcher.getCurrentScene().getClass(), MenuScene.class);
+    }
+
+    @Test
+    public void startGameWithAddingPlayers(){
+        //ARRANGE
+        var PLAYER_NAME = "testPlayer";
+
+        //ACT
+        window.button(TestComponentNames.NEW_PLAYER).click();
+        var optionPaneFixture = window.optionPane();
+        optionPaneFixture.textBox().enterText(PLAYER_NAME);
+        optionPaneFixture.okButton().click();
+        window.button(TestComponentNames.START_GAME).click();
+
+        //ASSERT
+        assertEquals(launcher.getCurrentScene().getClass(), GameScene.class);
+    }
+
+    @Test
+    public void reachedMaxPlayerCount(){
+        //ARRANGE
+        var PLAYER_NAME = "testPlayer";
```

Comment:
We could move this to the util folder and use it in the other test instead of 'alma'